src/components/TableWithSearchFilters/index.tsx

```diff
@@ -126,6 +128,16 @@ const TableWithSearchFilters = (props: TableWithSearchFiltersProps) => {
     });
   }, [extraTableFilters]);
 
+  const [tableDensity, setTableDensity] = useState<TableDensityType>();
```

Comment:
Is it ok that `tableDensity` is `undefined` by default? Do `TableDensitySettingsButton` and `OrderPreservedTable` both provide a default value when `density` is `undefined`?

Reply:
It is ok! The BaseTable will load tableDensity from userPreference by default. The state management is just to speed it up by like half a second, to update the density settings before the BE can confirm it basically.

Reply:
Ideally we move every table to use the TableWithSearchFilter component, and manage the state in one place only.